db/view/view.hh

```diff
@@ -123,6 +123,70 @@ bool matches_view_filter(const schema& base, const view_info& view, const partit
 
 bool clustering_prefix_matches(const schema& base, const partition_key& key, const clustering_key_prefix& ck);
 
+/**
+ * The liveness of rows resulting from an update to a table with an index over collection column
+ * depends on the liveness of the collection cells. bytes_with_action is used to return this information
+ * from the function computing the result for a particular computed column.
+ */
+struct bytes_with_action {
+    struct no_action {};
+    struct shadowable_tombstone_tag {
+        api::timestamp_type ts;
+        shadowable_tombstone into_shadowable_tombstone(gc_clock::time_point now) const {
+            return shadowable_tombstone{ts, now};
+        }
+    };
+
+    struct apply_visitor {
+        deletable_row* row;
+        gc_clock::time_point now;
+        bool apply_row_marker;
+        bool apply_tombstone;
+        bool operator()(no_action) {
+            return false;
+        }
+        bool operator()(shadowable_tombstone_tag t) {
+            if (apply_tombstone) {
+                row->apply(t.into_shadowable_tombstone(now));
+                return true;
+            }
+            return false;
+        }
+        bool operator()(row_marker rm) {
+            if (apply_row_marker) {
+                row->apply(rm);
+                return true;
+            }
+            return false;
+        }
+    };
+
+    struct action {
+        std::variant<no_action, row_marker, shadowable_tombstone_tag> _action = no_action{};
```

Comment:
nitpick: we use underscores for private/protected members of a class; if it's a struct and this _action is meant to be publicly accessed, just call it something like `value`. While `value` is a terrible name on its own, `action.value` is perfectly clear, and that's how it will be used.

Reply:
re: upper comment (for some reason I cannot reply to it) https://github.com/scylladb/scylla/pull/10123#discussion_r883549818

If anything, this corresponds to 
```
if (auto marker = std::get_if<row_marker>(&action._action; marker) {
    r->apply(*marker, now);
} else {
    r->apply(update_marker);
}
```
which perhaps is a bit more bloated than it seems. Either way, I can remove this abstraction.